contrib/repro_test.bzl

```diff
@@ -0,0 +1,243 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test rule for verifying a given container_image target produces a
+   deterministic Docker image.
+
+This rule builds the given container_image twice in isolation to determine if
+the given image is deterministic.
+It does so by mounting the provided Bazel project files into the image to build
+and reproduce the test image in (being the `base` attribute). Next, it
+sequentially runs two containers and builds the supplied container_image target
+(being `test_image`). For each container run, the rule extracts the output
+files for the test image. Finally, both extracted images are compared.
+A test image is considered to be reproducible when both instances of the same
+image have the same digest and ID. On a mismatch of one of those values, the
+test fails and the container_diff tool
+(https://github.com/GoogleContainerTools/container-diff#container-diff)
+produces the summary of their differences.
+
+Args:
+    test_image: A container_image target relative to the project's root to test
+                for determinism.
+    src_project_files: A target that exposes the files of the Bazel project to
+                       build the test_image in (e.g. filegroup target).
+    base: An image target to build and reproduce the test_image in.
+    container_diff_args: (optional) Args to the container_diff tool as specified here:
+                         https://github.com/GoogleContainerTools/container-diff#quickstart
+
+Example:
+
+container_repro_test(
+    name = "set_cmd_repro_test",
+    # Note that this target must always be relative to the project's root.
```

Comment:
Why is that restriction in place?

Reply:
Removed the restriction.
It was because I wasn't using Label.workspace_name, Label.package and Label.name